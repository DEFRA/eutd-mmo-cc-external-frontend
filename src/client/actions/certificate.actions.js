import { isError, showFullPageError } from './index';

export const GET_SUMMARY_DOCUMENT = 'GET_SUMMARY_DOCUMENT';
export const SHOW_SUMMARY_DOCUMENT_ERRORS = 'SHOW_SUMMARY_DOCUMENT_ERRORS';
export const GET_SUMMARY_UNAUTHORISED = 'GET_SUMMARY_DOCUMENT_UNAUTHORISED';

export function showSummaryCertificateErrors(data) {
  return { type: SHOW_SUMMARY_DOCUMENT_ERRORS, payload: { validationErrors: data } };
}

export const getSummaryCertificate = (journey, documentNumber = undefined) => async (dispatch, getState, { orchestrationApi }) => {
  const config = (documentNumber)
    ? { headers: { documentnumber: documentNumber } }
    : {};

  await orchestrationApi.get(`/certificate/${journey}`, config)
    .then(
      res => {
        dispatch({ type: GET_SUMMARY_DOCUMENT, payload: res.data });
      },
      error => {
        if (isError(error)) {
          dispatch(showFullPageError());
        } else {
          dispatch(unauthorised());
        }
      });

    function unauthorised() { return { type: GET_SUMMARY_UNAUTHORISED }; }
};