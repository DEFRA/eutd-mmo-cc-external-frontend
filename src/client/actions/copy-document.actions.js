import { showFullPageError, apiCallFailed, isError, CLEAR_ERRORS } from './index';
export const CONFIRM_COPY_DOCUMENT            = 'confirm_copy_document';
export const UNAUTHORISED_COPY_DOCUMENT       = 'unauthorised_copy_document';
export const CLEAR_COPY_DOCUMENT              = 'clear_copy_document';
export const REMOVE_CONFIRM_COPY_CERTIFICATE  = 'remove_confirm_copy_certificate';
export const ADD_ERROR_COPY_CERTIFICATE       = 'add_error_copy_certificate';
export const REMOVE_ERROR_COPY_CERTIFICATE    = 'remove_error_copy_certificate';

export const submitCopyCertificate = (payload) => async (dispatch, _getState, {orchestrationApi}) => {
  const {documentNumber, copyDocumentAcknowledged, voidOriginal, excludeLandings, journey} = payload;
  const config = {headers: {documentnumber: documentNumber}};

  try {
    let ipAddress;
    if (voidOriginal) {
      const ip = await orchestrationApi.get('/client-ip');
      ipAddress = ip.data;
    }

    return await orchestrationApi.post('/confirm-copy-certificate', { copyDocumentAcknowledged, voidOriginal, excludeLandings, journey, ipAddress }, config)
    .then(res => {
      dispatch({ type: CONFIRM_COPY_DOCUMENT, payload: {
        documentNumber: res.data.documentNumber,
        copyDocumentAcknowledged: res.data.copyDocumentAcknowledged,
        voidDocumentConfirm: res.data.voidOriginal
      }});
      return res.data.newDocumentNumber;
    });
  } catch (e) {
    if (isError(e)) dispatch(showFullPageError());
    if (e.response && e.response.status === 403) {
      dispatch({ type: UNAUTHORISED_COPY_DOCUMENT }); return;
    } else {
      dispatch(apiCallFailed(e.response));
    }
  }
};

export const checkCopyCertificate = (documentNumber) => async (dispatch, _getState, {orchestrationApi}) => {
  const config = {
    headers: {
      documentnumber: documentNumber
    }
  };

  try {
    const res = await orchestrationApi.get('/check-copy-certificate', config);
    const { canCopy } = res.data;
    if (!canCopy) {
      dispatch(unauthorised());
    }
  } catch(e) {
    if (isError(e)) dispatch(showFullPageError());
    if (e.response && e.response.status === 403) {
      dispatch(unauthorised());
      throw new Error(`Unauthorised access ${e.response.status}`);
    } else {
      dispatch(apiCallFailed(e.response));
      throw new Error(`An error has occurred ${e.response.status} (${e.response.statusText})`);
    }
  }

  function unauthorised() { return { type: UNAUTHORISED_COPY_DOCUMENT }; }
};

export const addCopyDocument = (data) => (dispatch) => dispatch({ type: CONFIRM_COPY_DOCUMENT, payload: data });
export const clearErrors = () => (dispatch) =>  dispatch({ type: CLEAR_ERRORS });
export const clearCopyDocument = () => (dispatch) => dispatch({ type: CLEAR_COPY_DOCUMENT });
export const unauthorisedCopyDocument = () => (dispatch) => dispatch({ type: UNAUTHORISED_COPY_DOCUMENT });