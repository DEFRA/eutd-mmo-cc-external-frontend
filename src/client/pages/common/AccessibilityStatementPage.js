import React from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { getAllUserAttributes } from '../../actions/userAttributes.actions';
import { AccessibilityStatementEnglish } from './AccessibilityStatementEnglish';
import { AccessibilityStatementWelsh } from './AccessibilityStatementWelsh';

const AccessibilityStatementPage = (props) => {
  let userSelectedLanguage;
  const userAttributeLanguage = props.userAttributes.find((val) => val.name === 'language');
  
  if (userAttributeLanguage) {
    userSelectedLanguage = userAttributeLanguage.value;
  }

  return (
    userSelectedLanguage === 'cy_UK' 
      ? <AccessibilityStatementWelsh />
      : <AccessibilityStatementEnglish />
  );
};

const mapStateToProps = state => {
  return {
    userAttributes: state.userAttributes
  };
};

const loadData = store => {
  return store.dispatch(getAllUserAttributes());
};

export default {
  loadData,
  component: withRouter(connect(mapStateToProps,
    {
      getAllUserAttributes
    })(AccessibilityStatementPage))
};
