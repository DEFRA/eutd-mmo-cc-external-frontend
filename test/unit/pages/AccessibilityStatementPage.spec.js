import { mount } from 'enzyme';
import * as React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import * as AccessibilityStatementPage from '../../../src/client/pages/common/AccessibilityStatementPage';
import AccessibilityStatementPageWrapper from '../../../src/client/pages/common/AccessibilityStatementPage';
import { getAllUserAttributes } from '../../../src/client/actions/userAttributes.actions';

jest.mock('../../../src/client/actions/userAttributes.actions');

describe('Accessibility Statement Page', () => {

  const mockStore = configureStore([thunk]);

  const getWrapper = (selectedLanguage) => {
    const store = mockStore({
      userAttributes: [{ name: 'language', value: selectedLanguage }],
    });

    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <AccessibilityStatementPage.default.component />
        </MemoryRouter>
      </Provider>
    );

    return wrapper;
  };

  it('should render the page in English if no language has been selected', () => {
    const store = mockStore({
      userAttributes: [],
    });

    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <AccessibilityStatementPage.default.component />
        </MemoryRouter>
      </Provider>
    );

    expect(wrapper).toBeDefined();
    expect(wrapper.find('PageTitle[title=\'Accessibility - GOV.UK\']').exists()).toBeTruthy();
  });

  it('should render the page in English', () => {
    const wrapper = getWrapper('en_UK');
    expect(wrapper).toBeDefined();
  });

  it('should render the page in Welsh', () => {
    const wrapper = getWrapper('cy_UK');
    expect(wrapper).toBeDefined();
  });

  it('should render PageTitle with the correct text in English', () => {
    const wrapper = getWrapper('en_UK');

    expect(wrapper.find('PageTitle[title=\'Accessibility - GOV.UK\']').exists()).toBeTruthy();
  });

  it('should render PageTitle with the correct text in Welsh', () => {
    const wrapper = getWrapper('cy_UK');

    expect(wrapper.find('PageTitle[title=\'Hygyrchedd - GOV.UK\']').exists()).toBeTruthy();
  });

  it('should take a snapshot of the whole page', () => {
    const wrapper = getWrapper();
    const { container } = render(wrapper);

    expect(container).toMatchSnapshot();
  });

  it('should call getAllUserAttributes action to load the component', () => {
    const store = { dispatch: jest.fn() };

    getAllUserAttributes.mockReturnValue({ type: 'GET_ALL_USER_ATTRIBUTES' });

    AccessibilityStatementPageWrapper.loadData(store);

    expect(getAllUserAttributes).toHaveBeenCalled();
    expect(getAllUserAttributes).toHaveBeenCalledTimes(1);
  });
});