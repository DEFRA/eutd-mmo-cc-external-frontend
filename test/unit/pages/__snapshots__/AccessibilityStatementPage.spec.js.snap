// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accessibility Statement Page should take a snapshot of the whole page 1`] = `
<div>
  <div
    class="css-1l99ixu-OuterContainer e4u1u2g0"
  >
    <div
      class="css-n57fe4-InnerContainer e4u1u2g1"
    >
      <div
        class="css-1lzr9ic-StyledRow e4u1u2g0"
      >
        <div
          class="css-1w80l5h-StyledColumn e4u1u2g0"
        >
          <h1
            class="e1vcssft0 css-11nzzut-StyledHeader-StyledHoc e4u1u2g0"
          >
            Accessibility statement for the Fish Exports service
          </h1>
          <p>
            This service is part of the wider 
            <a
              aria-label="Opens link for GOV.UK"
              href="https://www.gov.uk"
              rel="noopener noreferrer"
              target="_blank"
            >
              GOV.UK
              <span
                class="govuk-visually-hidden"
              >
                (opens in new tab)
              </span>
            </a>
             website. There’s a separate  
            <a
              aria-label="Opens link for accessibility statement for main gov.uk site"
              href="https://www.gov.uk/help/accessibility-statement"
              rel="noopener noreferrer"
              target="_blank"
            >
              accessibility statement for the main GOV.UK website.
              <span
                class="govuk-visually-hidden"
              >
                (opens in new tab)
              </span>
            </a>
          </p>
          <p>
            This page only contains information about the Fish Exports service, available at: 
            <br />
            <a
              aria-label="Opens link for information on fish export service"
              href="/"
              rel="noopener noreferrer"
              target="_blank"
            >
              https://manage-fish-exports.service.gov.uk/
              <span
                class="govuk-visually-hidden"
              >
                (opens in new tab)
              </span>
            </a>
          </p>
          <br />
          <h2
            class="e1vcssft0 css-1wlsbnr-StyledHeader-StyledHoc e4u1u2g0"
          >
            Using this service
          </h2>
          <p>
            This service is run by the Marine Management Organisation. We want as many people as possible to be able to use this service. For example, that means you should be able to:
          </p>
          <ul
            class="ul e1vcssft0 css-t0nsp0-StyledList-StyledHoc e4u1u2g0"
          >
            <li
              class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
            >
              change colours, contrast levels and fonts
            </li>
            <li
              class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
            >
              zoom in up to 300% without the text spilling off the screen
            </li>
            <li
              class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
            >
              get from the start of the service to the end using just a keyboard
            </li>
            <li
              class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
            >
              get from the start of the service to the end using speech recognition software
            </li>
            <li
              class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
            >
              listen to the service using a screen reader (including the most recent versions of JAWS, NVDA and VoiceOver)
            </li>
          </ul>
          <br />
          <p>
            We’ve also made the text in the service as simple as possible to understand.
          </p>
          <p>
            <a
              aria-label="Opens link about using your device if you have disability"
              href="https://mcmw.abilitynet.org.uk/"
              rel="noopener noreferrer"
              target="_blank"
            >
              AbilityNet
              <span
                class="govuk-visually-hidden"
              >
                (opens in new tab)
              </span>
            </a>
             has advice on making your device easier to use if you have a disability.
          </p>
          <h2
            class="e1vcssft0 css-1wlsbnr-StyledHeader-StyledHoc e4u1u2g0"
          >
            How accessible this service is
          </h2>
          <p>
            We know some parts of this website are not fully accessible.
          </p>
          <ul
            class="ul e1vcssft0 css-t0nsp0-StyledList-StyledHoc e4u1u2g0"
          >
            <li
              class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
            >
              There are compatibility issues with some assistive technology
            </li>
            <li
              class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
            >
              Some navigation is inconsistent or not clearly connected to its content
            </li>
            <li
              class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
            >
              There are text resizing issues on certain pages
            </li>
            <li
              class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
            >
              Accessible names are missing from some buttons
            </li>
            <li
              class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
            >
              Some error messages require clarification
            </li>
            <li
              class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
            >
              There is incorrect heading structure on certain pages
            </li>
            <li
              class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
            >
              There is an incorrect page title implementation on the Catch Certificates Add landings page
            </li>
            <li
              class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
            >
              Labels can become removed from their content on the Catch Certificates Add landings page
            </li>
          </ul>
          <br />
          <h2
            class="e1vcssft0 css-1wlsbnr-StyledHeader-StyledHoc e4u1u2g0"
          >
            Feedback and contact information
          </h2>
          <p>
            If you have difficulty using this service, contact us by:
          </p>
          <ul
            class="ul e1vcssft0 css-t0nsp0-StyledList-StyledHoc e4u1u2g0"
          >
            <li
              class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
            >
              call 
              <a
                href="tel:0330 159 1989 "
              >
                0330 159 1989
              </a>
            </li>
          </ul>
          <br />
          <p>
            As part of providing this service, we may need to send you messages or documents. We’ll ask you how you want us to send messages or documents to you but contact us if you need them in a different format. For example, large print, audio recording or braille.
          </p>
          <h2
            class="e1vcssft0 css-1wlsbnr-StyledHeader-StyledHoc e4u1u2g0"
          >
             Reporting accessibility problems with this service
          </h2>
          <p>
            We’re always looking to improve the accessibility of this service. If you find any problems that are not listed on this page or think we’re not meeting accessibility requirements, contact: Dom Horsfall (
            <a
              href="mailto:dominic.Horsfall@marinemanagement.org.uk"
            >
              dominic.Horsfall@marinemanagement.org.uk
            </a>
            )
          </p>
          <br />
          <h2
            class="e1vcssft0 css-1wlsbnr-StyledHeader-StyledHoc e4u1u2g0"
          >
            Enforcement procedure
          </h2>
          <p>
            The Equality and Human Rights Commission (EHRC) is responsible for enforcing the Public Sector Bodies (Websites and Mobile Applications) (No. 2) Accessibility Regulations 2018 (the ‘accessibility regulations’). If you’re not happy with how we respond to your complaint,  
            <a
              aria-label="Opens link to information about equality advisory service"
              href="https://www.equalityadvisoryservice.com/"
              rel="noopener noreferrer"
              target="_blank"
            >
              contact the Equality Advisory and Support Service (EASS).
              <span
                class="govuk-visually-hidden"
              >
                (opens in new tab)
              </span>
            </a>
          </p>
          <br />
          <h2
            class="e1vcssft0 css-1wlsbnr-StyledHeader-StyledHoc e4u1u2g0"
          >
            Contacting us by phone or visiting us in person
          </h2>
          <p>
            We provide a text relay service for people who are deaf, hearing impaired or have a speech impediment.
          </p>
          <p>
            Our offices have audio induction loops, or if you contact us before your visit, we can arrange a British Sign Language (BSL) interpreter to help you complete the service in person.
          </p>
          <p>
            Find out how to contact us:  
            <a
              aria-label="Opens link to Contact your local Marine Management Organisation office"
              href="https://www.gov.uk/contact-local-marine-management-organisation"
              rel="noopener noreferrer"
              target="_blank"
            >
              Contact your local Marine Management Organisation office
              <span
                class="govuk-visually-hidden"
              >
                (opens in new tab)
              </span>
            </a>
          </p>
          <br />
          <h2
            class="e1vcssft0 css-1wlsbnr-StyledHeader-StyledHoc e4u1u2g0"
          >
            Technical information about this service’s accessibility
          </h2>
          <p>
            The Marine Management Organisation is committed to making this service accessible, in accordance with the Public Sector Bodies (Websites and Mobile Applications) (No. 2) Accessibility Regulations 2018.
          </p>
          <br />
          <h3
            class="e1vcssft0 css-1wlsbnr-StyledHeader-StyledHoc e4u1u2g0"
          >
            Compliance status
          </h3>
          <p>
            This website is partially compliant with the 
            <a
              aria-label="Opens link to Web Content Accessibility Guidelines"
              href="https://www.w3.org/TR/WCAG21/"
              rel="noopener noreferrer"
              target="_blank"
            >
              Web Content Accessibility Guidelines version 2.1 AA standard
              <span
                class="govuk-visually-hidden"
              >
                (opens in new tab)
              </span>
            </a>
            , due to the non-compliances listed below.
          </p>
          <br />
          <h2
            class="e1vcssft0 css-1wlsbnr-StyledHeader-StyledHoc e4u1u2g0"
          >
            Non accessible content
          </h2>
          <p>
            The content listed below is non-accessible for the following reasons.
          </p>
          <br />
          <h3
            class="e1vcssft0 css-1wlsbnr-StyledHeader-StyledHoc e4u1u2g0"
          >
            Non-compliance with the accessibility regulations
          </h3>
          <p>
            The service has identified the following issues:
          </p>
          <table
            class="css-1uqv7s3-StyledTable e4u1u2g0"
          >
            <caption
              class="css-gzlitc-Caption e172z3n50"
            />
            <thead
              class="css-1iuscil-TableHeader e1tbxulj0"
            >
              <tr
                class="css-7u9vzt-RowInner e1bh3ka20"
              >
                <th
                  class="css-12e3svi-CellHeader e1sike1z0"
                >
                  Reference
                </th>
                <th
                  class="css-12e3svi-CellHeader e1sike1z0"
                >
                  Issue
                </th>
                <th
                  class="css-12e3svi-CellHeader e1sike1z0"
                >
                  Location
                </th>
                <th
                  class="css-12e3svi-CellHeader e1sike1z0"
                >
                  WCAG ref.
                </th>
              </tr>
            </thead>
            <tbody
              class="css-x1p7io-TableBody eo9ngsf0"
            >
              <tr
                class="css-7u9vzt-RowInner e1bh3ka20"
              >
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  <strong>
                    AA1/A-05a
                  </strong>
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  <strong>
                    Month-by-month navigation isn't visually or semantically connected to the 'Completed' data above it.
                  </strong>
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  CC,PS,SD - Exporter dashboard
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  1.3.2
                </td>
              </tr>
              <tr
                class="css-7u9vzt-RowInner e1bh3ka20"
              >
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  <strong>
                    AA1/A-05b
                  </strong>
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  <strong>
                    The '&lt;&lt;' and '&gt;&gt;' links to future/previous months are not explained or semantically presented in way that is accessible.
                  </strong>
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  CC,PS,SD - Exporter dashboard
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  1.3.2
                </td>
              </tr>
              <tr
                class="css-7u9vzt-RowInner e1bh3ka20"
              >
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  <strong>
                    AA1/A-23
                  </strong>
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  <strong>
                    Incorrect heading structure
                  </strong>
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  CC,PS,SD - Exporter dashboard
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  1.3.1
                </td>
              </tr>
              <tr
                class="css-7u9vzt-RowInner e1bh3ka20"
              >
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  <strong>
                    AA1/A-07
                  </strong>
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  <strong>
                    The text can be resized, but the way the form is handled means that the field widths don't adequately accommodate the resized text, partially obscuring the content.
                  </strong>
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  CC - Add landings
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  1.4.4
                </td>
              </tr>
              <tr
                class="css-7u9vzt-RowInner e1bh3ka20"
              >
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  <strong>
                    AA1/A-08
                  </strong>
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  <strong>
                    On text resizing, the form does not reflow on resizing, causing some content to be part hidden. Similarly, on smaller screen sizes, parts of the form appear off screen.
                  </strong>
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  CC - Add landings
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  1.4.10
                </td>
              </tr>
              <tr
                class="css-7u9vzt-RowInner e1bh3ka20"
              >
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  <strong>
                    AA1/A-09
                  </strong>
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  <strong>
                    Date picker issues.
                  </strong>
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  CC,PS,SD
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  2.1.1
                </td>
              </tr>
              <tr
                class="css-7u9vzt-RowInner e1bh3ka20"
              >
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  <strong>
                    AA1/A-10
                  </strong>
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  <strong>
                    Typeahead issues - ‘Add landings for each product’.
                  </strong>
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  CC,PS,SD
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  2.1.1
                </td>
              </tr>
              <tr
                class="css-7u9vzt-RowInner e1bh3ka20"
              >
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  <strong>
                    AA1/A-29
                  </strong>
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  <strong>
                    Page title issue with add landings - incorrect title element
                  </strong>
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  CC - Add landings
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  2.4.2
                </td>
              </tr>
              <tr
                class="css-7u9vzt-RowInner e1bh3ka20"
              >
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  <strong>
                    AA1/A-17
                  </strong>
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  <strong>
                    Accessible names are missing from the 'Edit' and 'Remove' buttons.
                  </strong>
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  CC - Add landings
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  2.5.3
                </td>
              </tr>
              <tr
                class="css-7u9vzt-RowInner e1bh3ka20"
              >
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  <strong>
                    AA1/A-30
                  </strong>
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  <strong>
                    Navigation issues - inconsistent main navigation
                  </strong>
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  CC,PS,SD - Exporter dashboard
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  3.2.3
                </td>
              </tr>
              <tr
                class="css-7u9vzt-RowInner e1bh3ka20"
              >
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  <strong>
                    AA1/A-18
                  </strong>
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  <strong>
                    Instances of generic error messaging.
                  </strong>
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  CC - Add landings
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  3.3.1
                </td>
              </tr>
              <tr
                class="css-7u9vzt-RowInner e1bh3ka20"
              >
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  <strong>
                    AA1/A-33
                  </strong>
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  <strong>
                    Landings form labels disappear from view when multiple landings are added.
                  </strong>
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  CC - Add landings
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  3.3.2
                </td>
              </tr>
              <tr
                class="css-7u9vzt-RowInner e1bh3ka20"
              >
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  <strong>
                    AA1/A-34
                  </strong>
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  <strong>
                    Validation is specific regarding “transportation details” but does not inform the user what they need to do to correct their mistakes.
                  </strong>
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  CC,SD - Add transportation details
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  3.3.1
                </td>
              </tr>
              <tr
                class="css-7u9vzt-RowInner e1bh3ka20"
              >
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  <strong>
                    AA1/A-01
                  </strong>
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  <strong>
                    Drop-down pick lists not accessible with Dragon Naturallyspeaking.
                  </strong>
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  CC - Add products
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  4.1.1
                </td>
              </tr>
              <tr
                class="css-7u9vzt-RowInner e1bh3ka20"
              >
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  <strong>
                    AA1/A-02
                  </strong>
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  <strong>
                    Date cannot be chosen using Dragon Naturallyspeaking.
                  </strong>
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  CC - Add landings
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  4.1.1
                </td>
              </tr>
              <tr
                class="css-7u9vzt-RowInner e1bh3ka20"
              >
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  <strong>
                    AA1/A-03
                  </strong>
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  <strong>
                    Drop-down pick lists not accessible with Dragon Naturallyspeaking.
                  </strong>
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  CC - Add landings
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  4.1.1
                </td>
              </tr>
              <tr
                class="css-7u9vzt-RowInner e1bh3ka20"
              >
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  <strong>
                    AA1/A-20
                  </strong>
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  <strong>
                    Compatibility issues with Dragon Naturally Speaking.
                  </strong>
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  CC,PS,SD
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  4.1.1
                </td>
              </tr>
              <tr
                class="css-7u9vzt-RowInner e1bh3ka20"
              >
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  <strong>
                    AA1/A-21
                  </strong>
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  <strong>
                    Navigation issues - missing ARIA roles
                  </strong>
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  CC,PS,SD
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  4.1.2
                </td>
              </tr>
              <tr
                class="css-7u9vzt-RowInner e1bh3ka20"
              >
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  <strong>
                    AA1/A-35
                  </strong>
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  <strong>
                    There is more than one type of format used for date pickers. This version is not keyboard-accessible.
                  </strong>
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  SD - Add product
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  4.1.1
                </td>
              </tr>
            </tbody>
          </table>
          <br />
          <p>
            CC = Catch Certificates, PS = Processing Statements, SD = Storage Documents
          </p>
          <p>
            We intend to fix these problems by as soon as possible.
          </p>
          <br />
          <h3
            class="e1vcssft0 css-1wlsbnr-StyledHeader-StyledHoc e4u1u2g0"
          >
            Disproportionate burden
          </h3>
          <p>
            For catch certificates:
          </p>
          <ul
            class="ul e1vcssft0 css-t0nsp0-StyledList-StyledHoc e4u1u2g0"
          >
            <li
              class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
            >
              the forms for direct landings are only available in PDF format
            </li>
          </ul>
          <br />
          <p>
            These PDF forms have passed WCAG AA standards, however, we do intend to replace them with HTML forms in the long term.
          </p>
          <h2
            class="e1vcssft0 css-1wlsbnr-StyledHeader-StyledHoc e4u1u2g0"
          >
            What we’re doing to improve accessibility
          </h2>
          <p>
            As part of continuous improvement, the service aims to:
          </p>
          <ul
            class="ul e1vcssft0 css-t0nsp0-StyledList-StyledHoc e4u1u2g0"
          >
            <li
              class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
            >
              address and fix identified, WCAG 2.1 AA non-compliant accessibility issues (priority 1)
            </li>
            <li
              class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
            >
              carry out continuous research to make sure you can complete the service first time, and make improvements where necessary (priority 2)
            </li>
            <li
              class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
            >
              continue to research and test the service to ensure your needs are addressed (priority 3)
            </li>
          </ul>
          <br />
          <h2
            class="e1vcssft0 css-1wlsbnr-StyledHeader-StyledHoc e4u1u2g0"
          >
            Preparation of this accessibility statement
          </h2>
          <p>
            This statement was prepared on 3 June 2021. It was last reviewed on 3 June 2021.
          </p>
          <p>
            This service was last tested on 21 September 2020. The test was carried out by Defra’s Digital, Data and Technology Services accessibility team.
          </p>
          <br />
        </div>
      </div>
    </div>
  </div>
</div>
`;
